metricbeat/module/vsphere/network/network_test.go

```diff
@@ -0,0 +1,93 @@
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package network
+
+import (
+	"testing"
+
+	mbtest "github.com/elastic/beats/v7/metricbeat/mb/testing"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/vmware/govmomi/simulator"
+)
+
+func TestFetchEventContents(t *testing.T) {
+	model := simulator.VPX()
+	if err := model.Create(); err != nil {
+		t.Fatal(err)
+	}
+
+	ts := model.Service.NewServer()
+	defer ts.Close()
+
+	f := mbtest.NewReportingMetricSetV2WithContext(t, getConfig(ts))
+	events, errs := mbtest.ReportingFetchV2WithContext(f)
+	if len(errs) > 0 {
+		t.Fatalf("Expected 0 error, had %d. %v\n", len(errs), errs)
+	}
+
+	assert.NotEmpty(t, events)
+
+	event := events[0].MetricSetFields
+
+	t.Logf("%s/%s event: %+v", f.Module().Name(), f.Name(), event.StringToPrint())
```

Comment:
This could be a bit more descriptive.

```suggestion
	t.Logf("Fetched event from %s/%s: %+v", f.Module().Name(), f.Name(), event)
```